tests/managing-issues/fixtures/run-provider-checks.py

```
@@ -99,665 +71,410 @@ def log_entries(path: Path) -> list[dict[str, Any]]:
+    require(json.loads(state_path.read_text())["fixtureWriteCount"] == 8, "Linear write count differs")
+
+
+def linear_edge_and_error(root: Path, base: dict[str, str]) -> None:
+    for scenario, fragment in (("absent", "unknown command"), ("incompatible", "incompatible")):
+        env, _, log = linear_env(root, base, f"linear-guide-{scenario}", MI_LINEAR_GUIDE_SCENARIO=scenario)
+        try:
+            load_linear_guide(env)
+        except (CheckFailure, json.JSONDecodeError) as error:
+            require(fragment in str(error).lower() or scenario == "absent", f"Linear {scenario} guide stopped for wrong reason")
+        else:
+            raise CheckFailure(f"Linear {scenario} guide unexpectedly enabled writes")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)